### PR TITLE
Add Dune project

### PR DESCRIPTION
**Pull request by LuAbelt**
Adds exemplary examples from Dune for compile-time variability (Template based)

Based on test cases in dune-pdelab

### CODE REVIEW

File: varats-core/varats/report/gnu_time_report.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def summary(self) -> str:
 
 class WLTimeReportAggregate(
     WorkloadSpecificReportAggregate[TimeReport],
-    shorthand=TimeReport.SHORTHAND + ReportAggregate.SHORTHAND,
+    shorthand="WL" + TimeReport.SHORTHAND + ReportAggregate.SHORTHAND,
```
Comment (boehmseb) [inline varats-core/varats/report/gnu_time_report.py:287]: What does the "WL" stand for?

File: varats/varats/projects/cpp_projects/dune.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            BinaryType.EXECUTABLE,
+            only_valid_in=separated_poisson_range
+        )
+
+        return binary_map[revision]
+
+    def compile(self) -> None:
+        """Compile the project using the in-built tooling from dune."""
+        version_source = local.path(self.source_of(self.primary_source))
+        self.cflags += ["-mllvm", "--vara-disable-phasar"]
```
Comment (boehmseb) [inline varats/varats/projects/cpp_projects/dune.py]: Why do you need this? Maybe add a small comment.
Also, why is this only needed here and not during recompile?

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats/varats/projects/cpp_projects/dune.py (hunk 2 above)
Comment (LuAbelt) [reply varats/varats/projects/cpp_projects/dune.py]: Unfortunately, the Phasar pass to annotate feature information currently crashes the compiler with Dune. 
So for now we are limited to analysing compile time features that do not require Phasar.

But good point, I should add a comment regarding this and I also forgot it for recompile

File: varats-core/varats/report/gnu_time_report.py (hunk 1 above)
Comment (LuAbelt) [reply varats-core/varats/report/gnu_time_report.py:287]: I introduced this as a shorthand for "Workload".

I added this as previously the `WLTimeReportAggregate` and `TimeReportAggregate` shared the same shorthand which caused issues for me when running the `TimeWorkloads` experiment.

File: varats/varats/projects/cpp_projects/dune.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    Simulation framework for various applications in mathematics and physics.
+
+    Note:
+        Currently Dune CANNOT be compiled with the Phasar passes activated
+        in vara.
+        Trying to do so will crash the compiler
+
+        If you use Dune with an experiment that uses the vara compiler,
+         add `-mllvm --vara-disable-phasar` to the projects `cflags` to disable phasar
+         passes.
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/dune.py]:
```suggestion
        If you use Dune with an experiment that uses the vara compiler,
        add `-mllvm --vara-disable-phasar` to the projects `cflags` to 
        disable phasar passes.
```

File: varats/varats/projects/cpp_projects/dune.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    """
+    Simulation framework for various applications in mathematics and physics.
+
+    Note:
+        Currently Dune CANNOT be compiled with the Phasar passes activated
+        in vara.
+        Trying to do so will crash the compiler
+
+        If you use Dune with an experiment that uses the vara compiler,
+        add `-mllvm --vara-disable-phasar` to the projects `cflags` to
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/dune.py]:
```suggestion
        add `-mllvm --vara-disable-phasar` to the projects `cflags` to
```

**Review by vulder — APPROVED** [review]

**Pull request merged**